api/client/proxy_test.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"context"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"testing"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+
+	"github.com/gravitational/teleport/lib/utils/socks"
+)
+
+// TestDialProxy tests the dialing mechanism of HTTP and SOCKS proxies.
+func TestDialProxy(t *testing.T) {
```

Comment:
Could you extend the https://github.com/gravitational/teleport/blob/1858aafa1569d836def5c2fdb06c0b0583777825/api/client/proxy/proxy_test.go#L32 test case and ensure that parsing socks5 HTTPS_PROXY logic as  with username and password (like `alice:password@socks5://proxy:1234`) is correctly.